src/plot_hourly_bar_graph.py

```diff
@@ -3,38 +3,70 @@
 import datetime
 import csv
 import numpy
+import os
+import copy
+
+directory = r"..\graphs\hourly_bar_graph"
+if not os.path.exists(directory):
+    os.makedirs(directory)
 
 list = []
-x = []
-y = []
-bins_x = []
 
 with open('../data/data.csv', 'r') as csvfile:
-    plots = csv.reader((l.replace('\0', '') for l in csvfile))
+    plots = csv.reader(csvfile, delimiter=',')
     for row in plots:
         try:
             list.append([datetime.datetime.strptime(
-                row[0], '%m/%d/%y %H:%M'), int(row[1])])
+                row[0], '%Y-%m-%d %H:%M:%S.%f'), int(row[1])])
         except:
             pass
 
-bins_x = []
-bins_y = []
+bins_x = set()
+date_dict = {}
```

Comment:
same as for hourly_line_graph.py but now you need both a `sensor_reading_cumu_day_map` and a `sensor_reading_noncumu_day_map`